src/charm.py

```diff
@@ -378,6 +384,20 @@ def _get_ue_subnet_config(self) -> Optional[str]:
     def _get_upf_core_ip_config(self) -> Optional[str]:
         return self.model.config.get("upf-core-ip")
 
+    def _multus_is_available(self) -> bool:
+        """Check whether Multus is enabled leveraging existence of NAD custom resource.
+
+        Returns:
+            bool: Whether Multus is enabled
+        """
+        client = Client()
+        try:
+            list(client.list(res=NetworkAttachmentDefinition, namespace=self.model.name))
+        except httpx.HTTPStatusError as e:
+            if e.response.status_code == 404:
+                return False
+        return True
```

Comment:
This should likely be placed outside of the class and take the namespace as a parameter, it's really a helper and does not really belong here imo. In fact can't we place it in the lib?